### ISSUE TITLE
EAS Update will fail if "runtimeVersion" is manually configured with something like "3.60"

### ISSUE THREAD

**Issue by pslandyl** [expo/eas-cli#1518]
### Summary

We manually specify the `"runtimeVersion"` in the `app.config.ts` file. Everything works fine util today when our version gets bump to `"3.60"`, the `eas update --branch dev --auto` command gives error.  

By changing `"runtimeVersion"` to something such as `"3.61"`, `"dev_3.60"`, or `"prod_3.60"` other than `"3.60"`, then everything works fine again. 

### Managed or bare?

Bare

### Environment

```
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.0 - /var/folders/64/fnh_hq_s7tjdc67k_p0tfp6r0000gn/T/fnm_multishells/3654_1666924208921/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.1.0 - /var/folders/64/fnh_hq_s7tjdc67k_p0tfp6r0000gn/T/fnm_multishells/3654_1666924208921/bin/npm
      Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /Users/thatsme/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        API Levels: 26, 27, 28, 29, 30, 31, 33
        Build Tools: 27.0.3, 30.0.0, 30.0.2, 30.0.3, 31.0.0, 33.0.0
        System Images: android-23 | Android TV ARM EABI v7a, android-29 | Android TV Intel x86 Atom, android-31 | Android TV ARM 64 v8a, android-31 | Google APIs ARM 64 v8a
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8609683
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ^0.3.9 => 0.3.9 
      @expo/webpack-config: ^0.17.0 => 0.17.2 
      expo: ^46.0.0 => 46.0.7 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.4 => 0.69.4 
      react-native-web: ~0.18.7 => 0.18.7 
    npmGlobalPackages:
      eas-cli: 2.7.0
    Expo Workflow: bare
```

### Error output

```
Failed to publish updates
    CombinedError: [GraphQL] Manifest Validation Error:
    runtimeVersion:'runtimeVersion' should be string beginning with an alphanumeric character followed by any combination of alphanumeric character, "_",
     "+", ".","(", ")", or "-". May not be a decimal ending in a 0. Valid examples: "1.0.3a+", "1.0.0", "._+()-0a1", "0".
    runtimeVersion:'runtimeVersion' should be a an 'exposdk:' prefix followed by the SDK version of your project. Example: "exposdk:44.0.0".
    runtimeVersion:should be object
    runtimeVersion:should match exactly one schema in oneOf
    This is likely a problem with your app.json, or app.config.js
```

### Reproducible demo or steps to reproduce from a blank project

1. in `app.config.ts` file, set the `runtimeVersion` to `"3.60"`
2. run command "eas update --branch dev --auto"

**Comment by antoinebrtd** [comment]
Had the same issue, try setting it to `3.60.0`

**Comment by thespacemanatee** [comment]
On Android, the version number gets coerced into an integer, so `3.60` becomes `3.6` and Expo has blocked such a versioning scheme. It's in the official docs somewhere.

**Comment by quinlanj** [comment]
hey @pslandyl - the reason why we don't allow runtime versions like `3.60` is for the reason that @thespacemanatee pointed out. On android, it coerces `3.60` to a float (`3.6`), instead of a string like the user intended. Here's the original issue for posterity: https://github.com/expo/expo/issues/16838

**Comment by angelEspada** [comment]
Would this issue happen if the version is `3.60.0`? or using semver avoids the version being coerced into float?

**Comment by quinlanj** [comment]
Update: We had actually fixed trailing `0` runtimeVersions after we identified the issue: https://github.com/expo/expo/pull/19821

I've manually tested Android runtimeVersions with trailing 0's in SDK48+ and confirm we no longer coerce to a float. As such, we will drop the 'trailing 0' restriction from SDK 48 and onwards. This change will be deployed to our servers later today. 

Sorry for all the confusion!

**Comment by quinlanj** [comment]
@angelEspada : Before this was fixed, your particular example of `3.60.0` should remain a string, but other cases like `3.60` were coerced to be floats in Android. This should no longer be an issue anymore and we will be lifting the restriction from our servers soon.
